no-restricted-syntax false positive

* **ESLint Version:** 4.0.0
* **Node Version:** 8.1.1
* **npm Version:** 5.0.3

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
This template is used for finding certain code patterns (`RegExp` with flags in this case)
```js
'use strict';

module.exports = {
  root: true,
  parserOptions: { ecmaVersion: 2017, sourceType: 'script' },
  env: { es6: true, node: true, browser: true },

  rules: {
    'no-restricted-syntax': ['error', 'Literal[regex.flags=/./]'],
  },
};
```

**What did you do? Please include the actual source code causing the issue.**
```js
'use strict';

console.log(/a/);
console.log(/a/i);
```

**What did you expect to happen?**
```console
 4:13  error  Using 'Literal[regex.flags=/./]' is not allowed  no-restricted-syntax
```

**What actually happened? Please include the actual, raw output from ESLint.**
```console
  1:1   error  Using 'Literal[regex.flags=/./]' is not allowed  no-restricted-syntax
  4:13  error  Using 'Literal[regex.flags=/./]' is not allowed  no-restricted-syntax
```

--- a/package.json
+++ b/package.json
@@ -44,7 +44,7 @@
     "eslint-scope": "^3.7.1",
     "eslint-visitor-keys": "^1.0.0",
     "espree": "^4.0.0-alpha.0",
-    "esquery": "^1.0.0",
+    "esquery": "^1.0.1",
     "esutils": "^2.0.2",
     "file-entry-cache": "^2.0.0",
     "functional-red-black-tree": "^1.0.1",

--- a/tests/lib/rules/no-restricted-syntax.js
+++ b/tests/lib/rules/no-restricted-syntax.js
@@ -37,9 +37,9 @@ ruleTester.run("no-restricted-syntax", rule, {
             code: "({ foo: 1, bar: 2 })",
             options: [{ selector: "FunctionDeclaration[params.length>2]", message: "custom error message." }]
         },
-        
+
         // https://github.com/eslint/eslint/issues/8733
-        { code: "console.log(/a/);", options: ["Literal[regex.flags=/./]"] },
+        { code: "console.log(/a/);", options: ["Literal[regex.flags=/./]"] }
     ],
     invalid: [
 
@@ -123,7 +123,7 @@ ruleTester.run("no-restricted-syntax", rule, {
             options: [{ selector: "FunctionDeclaration[params.length>2]", message: "custom message with {{selector}}" }],
             errors: [{ message: "custom message with {{selector}}", type: "FunctionDeclaration" }]
         },
-        
+
         // https://github.com/eslint/eslint/issues/8733
         {
             code: "console.log(/a/i);",

--- a/tests/lib/rules/no-restricted-syntax.js
+++ b/tests/lib/rules/no-restricted-syntax.js
@@ -98,7 +98,6 @@ ruleTester.run("no-restricted-syntax", rule, {
             options: ["FunctionDeclaration[params.length>2]"],
             errors: [{ message: "Using 'FunctionDeclaration[params.length>2]' is not allowed.", type: "FunctionDeclaration" }]
         },
-        { code: "console.log(/a/);", options: ["Literal[regex.flags=/./]"] },
 
         // object format
         {

--- a/tests/lib/rules/no-restricted-syntax.js
+++ b/tests/lib/rules/no-restricted-syntax.js
@@ -36,7 +36,10 @@ ruleTester.run("no-restricted-syntax", rule, {
         {
             code: "({ foo: 1, bar: 2 })",
             options: [{ selector: "FunctionDeclaration[params.length>2]", message: "custom error message." }]
-        }
+        },
+        
+        // https://github.com/eslint/eslint/issues/8733
+        { code: "console.log(/a/);", options: ["Literal[regex.flags=/./]"] },
     ],
     invalid: [
 
@@ -95,6 +98,7 @@ ruleTester.run("no-restricted-syntax", rule, {
             options: ["FunctionDeclaration[params.length>2]"],
             errors: [{ message: "Using 'FunctionDeclaration[params.length>2]' is not allowed.", type: "FunctionDeclaration" }]
         },
+        { code: "console.log(/a/);", options: ["Literal[regex.flags=/./]"] },
 
         // object format
         {
@@ -118,6 +122,13 @@ ruleTester.run("no-restricted-syntax", rule, {
             code: "function foo(bar, baz, qux) {}",
             options: [{ selector: "FunctionDeclaration[params.length>2]", message: "custom message with {{selector}}" }],
             errors: [{ message: "custom message with {{selector}}", type: "FunctionDeclaration" }]
+        },
+        
+        // https://github.com/eslint/eslint/issues/8733
+        {
+            code: "console.log(/a/i);",
+            options: ["Literal[regex.flags=/./]"],
+            errors: [{ message: "Using 'Literal[regex.flags=/./]' is not allowed.", type: "Literal" }]
         }
     ]
 });

